torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -314,3 +314,52 @@ def resized_crop_image_pil(
 ) -> PIL.Image.Image:
     img = crop_image_pil(img, top, left, height, width)
     return resize_image_pil(img, size, interpolation=interpolation)
+
+
+def random_pad_image_tensor(
```

Comment:
Why do we need this? Shouldn't `pad_image_tensor` be able to handle this? In general, we don't have kernels for random functions. All randomness should be handled in the transform.

Reply:
Hi @pmeier, sorry I am a bit new to this repo. The reason I created a new function is that:
1. `random_pad_image_tensor` does not have the same logic as this function. This function takes into account the output shape required and the current image shape. I wasn't sure if this logic should be in `forward`.
2. Seeing the current code, I felt that transform-related code should be in `_geometry` and that there should be separate functions for tensor and PIL.
3. `_get_params` in `RandomCrop` requires the output from this function. So adding this logic within `_transform` wouldn't work as the `params` in `_transform` would not be valid.

Please do provide me with any other approach you have in mind. I could incorporate those changes.

Reply:
Sorry, I misjudged the situation. I was not aware that the forward actually modified the image:

https://github.com/pytorch/vision/blob/b4cb352c586ee6104a79b2d367d019ca480759b3/torchvision/transforms/transforms.py#L663-L674

This makes things more complicated. cc @datumbox for awareness.

I would move this code into `_transform`. Although the structure is the same for all possible types, we still need to call different `pad` kernels. That would be a lot easier if we had the `Pad` transform from #5521 first. This way we could simply substitute `pad_image_*(...)` with `pad(...)` where `pad` is

```py
pad = functools.partial(
    lambda image, padding: Pad(
        padding,
        fill=self.fill,
        padding_mode=self.padding_mode,
    )(image)
)
``` 

and not worry about the dispatch. Thoughts?

Reply:
hi @pmeier , I can keep this PR on hold and work on #5521 first, if it helps.